hypothesis-python/src/hypothesis/core.py

```diff
@@ -1104,12 +1104,11 @@ def _execute_once_for_engine(self, data: ConjectureData) -> None:
             if TESTCASE_CALLBACKS:
                 if runner := getattr(self, "_runner", None):
                     phase = runner._current_phase
-                elif (
-                    self.failed_normally or self.failed_due_to_deadline
-                ):  # pragma: no cover  # FIXME
-                    phase = "shrink"
                 else:  # pragma: no cover  # in case of messing with internals
```

Comment:
Presumably this "messing with internals" is not totally unexpected, otherwise we wouldn't have code to handle it. But I couldn't find any tests that enter this method without `_runner` set.

We *could* consider just dropping this handling, and require that whoever messes also attaches a mock runner?